### ISSUE TITLE
Bug: SARIMA causes program to exit when updating results without refit

### ISSUE THREAD

**Issue by jelc53** [statsmodels/statsmodels#8350]
#### Describe the bug

SARIMA with exogenous variable input (fourier terms prepared with pmdarima's FourierFeaturizer) seems to be able to fit model parameters (training), but when I try to update the results object without refitting `results.append(updated_endog, exog=updated_exog, refit=False)` all hell breaks loose (kernel dies if working in jupyter, debugger exits without error msg in vscode).

#### Code Sample

Data is a csv with two columns: `interval` which is a timestamp and `fixed_load` which is a float. Each row represents a 30 min time interval, starting at the timestamped value in the `interval` column. I unfortunately cannot share the data for sensitivity reasons, but here is a sanitized sample.

```
                   interval   fixed_load
0 2022-03-01 10:00:00+00:00  2150
1 2022-03-01 10:30:00+00:00  2540
2 2022-03-01 11:00:00+00:00  2665
3 2022-03-01 11:30:00+00:00  2620
4 2022-03-01 12:00:00+00:00  2240
```

```python
import os
import numpy as np
import pandas as pd

from statsmodels.tsa.statespace.sarimax import SARIMAX
from pmdarima.preprocessing import FourierFeaturizer

def walk_forward_prediction(data, exog, n_steps, results, pred_start_idx, intervals_per_day=48):
    forecasts = {}
    
    for i in range(n_steps):
        
        # resolve indices
        start_idx = pred_start_idx + i*intervals_per_day
        end_idx = pred_start_idx + (i+1)*intervals_per_day - 1
        
        # update results by appending next observation
        updated_endog = data.loc[start_idx:end_idx, 'fixed_load']
        updated_exog = exog.loc[start_idx:end_idx,]
        results = results.append(updated_endog, exog=updated_exog, refit=False)
        
        # save the new set of forecasts
        forecasts[updated_endog.index[0]] = results.forecast(
            steps=intervals_per_day, 
            exog=updated_exog
        )
        
    # combine all forecasts into a dataframe
    forecasts = pd.concat(forecasts, axis=1)
    
    return forecasts

if __name__ == '__main__':
    print('load training data')
    data = pd.read_csv(
        os.path.join(input_dir, 'practice_dataset.csv'),
        parse_dates=['interval']
    )
    load_data = data['fixed_load']
    time_intervals = data['interval']

    print('split into test and train datasets')
    forecast_horizon = 48
    pred_start_idx = 336*8
    train_data = load_data[:pred_start_idx]
    test_data = load_data[pred_start_idx:(pred_start_idx+336)]
    # train_data, test_data = temporal_train_test_split(y=load_data, test_size=0.2)

    print('training model ...')
    # prepare the fourier terms to add as exogenous features
    four_terms = FourierFeaturizer(336, 1)
    y_prime, exog = four_terms.fit_transform(np.log(load_data))

    # split the time series as well as exogenous features data into train and test splits 
    # train_exog, test_exog = temporal_train_test_split(exog, test_size=0.2)
    train_exog = exog[:pred_start_idx]
    test_exog = exog[pred_start_idx:(pred_start_idx+336)]

    y_train_data = data[:pred_start_idx]
    y_train_data = np.log(y_train_data['fixed_load'])

    model_sarima_fourier = SARIMAX(
        endog=y_train_data,
        exog=train_exog,
        trend='n',
        order=(2, 0, 1),
        seasonal_order=(2, 0, 2, 48),  
    )
    results_sarima_fourier = model_sarima_fourier.fit()
    print(results_sarima_fourier.summary())

    print('walk forward forecasting ...')
    n_steps = 7
    walk_forward_forecast = walk_forward_prediction(
        data=data.copy(),
        exog=exog.copy(),
        n_steps=n_steps,
        results=results_sarima_fourier,
        pred_start_idx=pred_start_idx
    )
    print(walk_forward_forecast)


```
<details>

**Note**: As you can see, there are many issues on our GitHub tracker, so it is very possible that your issue has been posted before. Please check first before submitting so that we do not have to handle and close duplicates.

**Note**: Please be sure you are using the latest released version of `statsmodels`, or a recent build of `main`. If your problem has been fixed in an unreleased version, you might be able to use `main` until a new release occurs. 

**Note**: If you are using a released version, have you verified that the bug exists in the main branch of this repository? It helps the limited resources if we know problems exist in the current main branch so that they do not need to check whether the code sample produces a bug in the next release.

</details>

If the issue has not been resolved, please file it in the issue tracker.

#### Expected Output

I expected to get back a pandas series object `walk_forward_forecast` that I could print to console and perfom out-of-sample fit evaluations on.

#### Output of ``import statsmodels.api as sm; sm.show_versions()``

<details>

INSTALLED VERSIONS
------------------
Python: 3.7.13.final.0
OS: Linux 5.14.0-1045-oem #51-Ubuntu SMP Mon Jul 4 06:41:22 UTC 2022 x86_64
byteorder: little
LC_ALL: None
LANG: en_AU.UTF-8

statsmodels
===========

Installed: 0.13.2 (/home/julian/anaconda3/envs/sarimax/lib/python3.7/site-packages/statsmodels)

Required Dependencies
=====================

cython: 0.29.30 (/home/julian/anaconda3/envs/sarimax/lib/python3.7/site-packages/Cython)
numpy: 1.21.5 (/home/julian/anaconda3/envs/sarimax/lib/python3.7/site-packages/numpy)
scipy: 1.7.3 (/home/julian/anaconda3/envs/sarimax/lib/python3.7/site-packages/scipy)
pandas: 1.3.5 (/home/julian/anaconda3/envs/sarimax/lib/python3.7/site-packages/pandas)
    dateutil: 2.8.2 (/home/julian/anaconda3/envs/sarimax/lib/python3.7/site-packages/dateutil)
patsy: 0.5.2 (/home/julian/anaconda3/envs/sarimax/lib/python3.7/site-packages/patsy)

Optional Dependencies
=====================

matplotlib: 3.1.2 (/home/julian/anaconda3/envs/sarimax/lib/python3.7/site-packages/matplotlib)
    backend: module://matplotlib_inline.backend_inline 
cvxopt: Not installed
joblib: 1.1.0 (/home/julian/anaconda3/envs/sarimax/lib/python3.7/site-packages/joblib)

Developer Tools
================

IPython: 7.31.1 (/home/julian/anaconda3/envs/sarimax/lib/python3.7/site-packages/IPython)
    jinja2: Not installed
sphinx: Not installed
    pygments: 2.11.2 (/home/julian/anaconda3/envs/sarimax/lib/python3.7/site-packages/pygments)
pytest: Not installed
virtualenv: Not installed

</details>

**Comment by jelc53** [comment]
Follow up: I think it is to do with ``append`` method logic from ``MLEResults`` class. Looks like it only appends new observations to ``orig_endog`` or ``orig_exog`` and so might not work for mutliple function calls.

**Comment by ChadFulton** [comment]
Thanks very much for this report @jelc53.  It may be a week or so before I'll have a chance to look into this in depth and create a patch to fix whatever the issue is.  But if you happen to figure it out before then, that will be great too.

**Comment by mboldin** [comment]
I tried your code using made up data (see code below) and did NOT get a case of 'kernel dies if working in jupyter'

By simplifying the AR & MA terms to these 
    order=(1, 0, 0),
    seasonal_order=(0, 0, 1, 48),  
the walk_forward_prediction( ) steps  seemed to run fine

Your model spec 
    order=(2, 0, 1),
    seasonal_order=(2, 0, 2, 48),  
took a long time to estimate and started with this message
warn('Non-stationary starting autoregressive parameters'

This may  be due to the fact that my made up data has no AR, MA or seasonal effects.

Still I suspect your model really does not fit the data well or has a problem in some dimension. I.e. I suspect  a data or model specific case -- and not a general SARIMAX  problem

Can you try these cases with your data to see if the problem might lie in the model specification for your data, (and hence is hard to replicate)

1. Run with no exog data and order=(1, 0, 0), seasonal_order=(0, 0, 1, 48),  
2. Run with exog data and order=(1, 0, 0), seasonal_order=(0, 0, 1, 48),  
3. Run with no exog data and your preferred order=, seasonal_order= 
4. Run with exog data and your preferred order=, seasonal_order=

You could also try a simple exog data case.  Deriving  exog from thel oad_data as below
y_prime, exog = four_terms.fit_transform(np.log(load_data)) 
seems problematic since  y is also based on load_data 

-------------------------------------------------

My data creation steps 

np.random.seed(12345)

N = 336*8 + 10*48

d = pd.date_range('2022-03-01 10:00:00', periods = N, freq ='.5H')
data = pd.DataFrame(d)

y = np.random.uniform(2200,2600,size=N)
y = [ int(x) for x in y]
data['fixed_load'] = y

data.columns = 'interval  fixed_load'.split()

load_data = data['fixed_load']
time_intervals = data['interval']
``

**Comment by jelc53** [comment]
Hmm interesting, it did work for me when I tried these stripped down model formulations. Weird because the formulation I mentioned in the original comment is what performed best when using an ``msarima`` (multiple seasonal arima) model from r's ``smooth`` package so I don't think it is a fit thing so much as a training timeout issue.

**Comment by ChadFulton** [comment]
Thanks very much @mboldin for looking into creating an example dataset to replicate the issue.  I also wasn't able to replicate it. That doesn't mean there's not a problem somewhere, but I'm afraid we don't have enough to go on here, so I'm going to close this issue for now.  But @jelc53, if you do find an example with data that you can post, please reopen the issue.
